### PR TITLE
Add localization for UI strings in German

### PR DESCRIPTION
**Pull request by stepankuzmin**
Hey 👋

This PR adds translations for UI strings we use in GL JS into German. We translated the UI strings automatically using the Google Translate API, and we need to verify that the translation is correct.

To help us verify the translations, you can open the `Files changed` tab on this page and compare the translations with the original file [`src/ui/default_locale.js`](https://github.com/mapbox/mapbox-gl-js/blob/main/src/ui/default_locale.js).

If the translation looks good, you can approve it using the `Review changes` button in the `Files changed` tab and selecting the `Approve` radio button when submitting the review.

To improve the machine translation, you can click the plus sign next to the line you want to enhance and select the suggestion button on the toolbar.

After making the suggestions, you can mark the review as finished using the `Review changes` button and selecting the `Request changes` radio button.

Thanks for helping 🙌

### CODE REVIEW

File: src/ui/locales/de.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    "AttributionControl.MapFeedback": "Karten-Feedback",
+    "FullscreenControl.Enter": "Vollbildmodus aktivieren",
+    "FullscreenControl.Exit": "Vollbild beenden",
+    "GeolocateControl.FindMyLocation": "Finden Sie meinen Standort",
+    "GeolocateControl.LocationNotAvailable": "Ort nicht verfügbar",
+    "LogoControl.Title": "Mapbox-Logo",
+    "Map.Title": "Karte",
+    "NavigationControl.ResetBearing": "Setzen Sie die Peilung auf Nord zurück",
+    "NavigationControl.ZoomIn": "Hineinzoomen",
+    "NavigationControl.ZoomOut": "Rauszoomen",
```
Comment (melanieimfeld) [inline src/ui/locales/de.js:14]: I think this translation is an abbreviation. Suggestion: `herauszoomen`

File: src/ui/locales/de.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+const locale = {
+    "AttributionControl.ToggleAttribution": "Zuordnung umschalten",
+    "AttributionControl.MapFeedback": "Karten-Feedback",
+    "FullscreenControl.Enter": "Vollbildmodus aktivieren",
+    "FullscreenControl.Exit": "Vollbild beenden",
+    "GeolocateControl.FindMyLocation": "Finden Sie meinen Standort",
+    "GeolocateControl.LocationNotAvailable": "Ort nicht verfügbar",
+    "LogoControl.Title": "Mapbox-Logo",
+    "Map.Title": "Karte",
+    "NavigationControl.ResetBearing": "Setzen Sie die Peilung auf Nord zurück",
```
Comment (melanieimfeld) [inline src/ui/locales/de.js:12]:
```suggestion
    "NavigationControl.ResetBearing": "Setzen Sie die Peilung auf Norden zurück",
```

File: src/ui/locales/de.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    "NavigationControl.ZoomIn": "Hineinzoomen",
+    "NavigationControl.ZoomOut": "Rauszoomen",
+    "ScaleControl.Feet": "ft",
+    "ScaleControl.Meters": "m",
+    "ScaleControl.Kilometers": "km",
+    "ScaleControl.Miles": "mich",
```
Comment (melanieimfeld) [inline src/ui/locales/de.js:18]:
```suggestion
    "ScaleControl.Miles": "mi",
```

File: src/ui/locales/de.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+// @flow
+
+const locale = {
+    "AttributionControl.ToggleAttribution": "Zuordnung umschalten",
```
Comment (melanieimfeld) [inline src/ui/locales/de.js:4]: Does this refer to map attribution? If so, I don't think `Zuordung` is an ideal translation here. [Esri](https://support.esri.com/de/technical-article/000012040) uses `Attributierung`

```suggestion
    "AttributionControl.ToggleAttribution": "Attributierung umschalten",
```

File: src/ui/locales/de.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    "AttributionControl.MapFeedback": "Karten-Feedback",
+    "FullscreenControl.Enter": "Vollbildmodus aktivieren",
+    "FullscreenControl.Exit": "Vollbild beenden",
```
Comment (sebrockm) [inline src/ui/locales/de.js:7]: I think this should be consistent with the one above. So, `Vollbildmodus beenden` or even more consistent `Vollbildmodus deaktivieren`. Alternatively, remove `modus` in both lines.

File: src/ui/locales/de.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    "GeolocateControl.FindMyLocation": "Finden Sie meinen Standort",
```
Comment (sebrockm) [inline src/ui/locales/de.js:8]:
```suggestion
    "GeolocateControl.FindMyLocation": "Finde meinen Standort",
```

I think this sentence is thought to be said by a human towards a machine? We don't say "Alexa, stellen Sie einen Timer", either.

File: src/ui/locales/de.js (hunk 2 above)
Comment (sebrockm) [reply src/ui/locales/de.js:12]: Same as above, I think this is a human giving instructions to a machine, so, aggregating the previous comments, I suggest

``` suggestion
    "NavigationControl.ResetBearing": "Setze die Orientierung auf Norden zurück",
```

File: src/ui/locales/de.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    "GeolocateControl.LocationNotAvailable": "Ort nicht verfügbar",
+    "LogoControl.Title": "Mapbox-Logo",
+    "Map.Title": "Karte",
+    "NavigationControl.ResetBearing": "Setzen Sie die Peilung auf Nord zurück",
+    "NavigationControl.ZoomIn": "Hineinzoomen",
+    "NavigationControl.ZoomOut": "Rauszoomen",
+    "ScaleControl.Feet": "ft",
```
[the remark this answers is not in the document]
Comment (sebrockm) [reply src/ui/locales/de.js:15]: True, but I think that is due to the fact that it is this unit which is not commonly used in the first place, so it's abbreviation isn't either. **If** however, "Fuß" happens to be used as a unit in German, then I think `ft` would be the correct abbreviation. Actually, abbreviation is not the correct term here, I think. It rather is a symbol that happens to be an abbreviation in this case. Consider e.g. `cm` or `µm` which are not really abbreviations for **Z**entimeter or **Mi**krometer. [German Wikipedia](https://de.wikipedia.org/wiki/Fu%C3%9F_(Einheit)) also uses `ft` and calls it "Einheitenzeichen".

**Review by sebrockm — APPROVED** [review]

**Pull request merged**